CONTRIBUTING.md

```diff
@@ -1,197 +1,221 @@
-
-As contributors and maintainers to this project,
-you are expected to abide by cuDF's code of conduct.
-More information can be found at: [Contributor Code of Conduct](https://docs.rapids.ai/resources/conduct/).
+Contributions to cuDF fall into the following categories:
+
+1. To report a bug, request a new feature, or report a problem with documentation, please file an
+   [issue](https://github.com/rapidsai/cudf/issues/new/choose) describing the problem or new feature
+   in detail. The RAPIDS team evaluates and triages issues, and schedules them for a release. If you
+   believe the issue needs priority attention, please comment on the issue to notify the team.
+2. To propose and implement a new feature, please file a new feature request
+   [issue](https://github.com/rapidsai/cudf/issues/new/choose). Describe the intended feature and
+   discuss the design and implementation with the team and community. Once the team agrees that the
+   plan looks good, go ahead and implement it, using the [code contributions](#code-contributions)
+   guide below.
+3. To implement a feature or bug fix for an existing issue, please follow the [code
+   contributions](#code-contributions) guide below. If you need more context on a particular issue,
+   please ask in a comment.
+
+As contributors and maintainers to this project, you are expected to abide by cuDF's code of
+conduct. More information can be found at:
+[Contributor Code of Conduct](https://docs.rapids.ai/resources/conduct/).
 
 ## Code contributions
 
 ### Your first issue
 
-1. Follow the guide at the bottom of this page for [Setting Up Your Build Environment](#setting-up-your-build-environment).
-2. Find an issue to work on. The best way is to look for the [good first issue](https://github.com/rapidsai/cudf/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
-    or [help wanted](https://github.com/rapidsai/cudf/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) labels.
+1. Follow the guide at the bottom of this page for
+   [Setting up your build environment](#setting-up-your-build-environment).
+2. Find an issue to work on. The best way is to look for the
+   [good first issue](https://github.com/rapidsai/cudf/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+   or [help wanted](https://github.com/rapidsai/cudf/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
+   labels.
 3. Comment on the issue stating that you are going to work on it.
-4. Code! Make sure to update unit tests!
-5. When done, [create your pull request](https://github.com/rapidsai/cudf/compare).
-6. Verify that CI passes all [status checks](https://help.github.com/articles/about-status-checks/). Fix if needed.
-7. Wait for other developers to review your code and update code as needed.
-8. Once reviewed and approved, a RAPIDS developer will merge your pull request.
+4. Write code to address the issue or implement the feature.
```

Comment:
it might help some newer contributors to clarify that this is supposed to happen in a branch in their own fork.

Reply:
That's a very good catch. This is commonly forgotten and results in PRs from the branch `username/branch-22.06`, which gets messy...